### PR TITLE
feat: adds unsubscribe button and better templates

### PR DESCRIPTION
**Pull request by fcanela**
Requires a SMTP email to handle unsubscriptions. Adds Plinio email templates.

## Release Notes

- **New Features**
  - Introduced an unsubscribe email option for notifications, allowing users to manage their email preferences.

- **Improvements**
  - Enhanced email templates for better readability and user experience, including modular components and improved styling.
  - Updated documentation for clearer configuration guidance related to new email management features.

- **Bug Fixes**
  - Minor formatting adjustments for better compliance with email standards.

### CODE REVIEW

File: src/utils/notification-template.ts [hunk 1]
```diff
@@ -50,11 +52,12 @@ export const render = (transport: TransportName, notification: Notification, var
     formatAnswer,
     getRealityQuestionLink,
     getSnapshotProposalLink,
+    unsubscribeEmail,
   };
 
   return ejs.renderFile(path, context, {
     cache: true,
-    async: true,
+    async: false,
```
Comment (fcanela) [inline src/utils/notification-template.ts:60]: This caused me some headache. It theorically should not be required, but somehow if I let the async happen, the rendered result contains `[Object promise]` in every place an EJS `include` happens. The code that is calling this function has an await, so I have no clue of what is causing this behavior. Forcing synchronous rendering fixed it.

File: templates/email/answer-issued-html.ejs [hunk 2]
```diff
@@ -1,80 +1,67 @@
...
-  </body>
-</html>
\ No newline at end of file
+<%- include('./components/layout-start', { title: 'New Reality Answer'}) %>
+
+<table width="100%" cellspacing="0" cellpadding="0" role="presentation" style="mso-table-lspace:0pt;mso-table-rspace:0pt;border-collapse:collapse;border-spacing:0px">
+    <tr>
+        <td align="left" style="padding:0;Margin:0;padding-top:25px;padding-bottom:40px">
+            <h1 style="Margin:0;line-height:25.2px;mso-line-height-rule:exactly;font-family:'open sans', 'helvetica neue', helvetica, arial, sans-serif;font-size:21px;font-style:normal;font-weight:normal;color:#daf0ff">Hi there!</h1>
+            <h1 style="Margin:0;line-height:25.2px;mso-line-height-rule:exactly;font-family:'open sans', 'helvetica neue', helvetica, arial, sans-serif;font-size:21px;font-style:normal;font-weight:normal;color:#daf0ff">A new Reality answer has been submitted for a Snapshot proposal in the space 1inch.eth. </h1>
```
Comment (fcanela) [inline templates/email/answer-issued-html.ejs:7]: I forgot to change the `1inch.eth` value for the ENS. I'm fixing this.

File: templates/email/components/layout-end.ejs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                                                        </table>
+                                                    </td>
+                                                </tr>
+                                            </table>
+                                        </td>
+                                    </tr>
+                                </table>
+                            </td>
+                        </tr>
+                    </table>
```
[the remark this answers is not in the document]
Comment (fcanela) [reply templates/email/components/layout-end.ejs:34]: This is an EJS template. That value will be replaced by the real link on rendering.

File: templates/email/components/layout-start.ejs [hunk 4]
```diff
@@ -0,0 +1,145 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html dir="ltr" xmlns="http://www.w3.org/1999/xhtml" xmlns:o="urn:schemas-microsoft-com:office:office" lang="en" style="font-family:arial, 'helvetica neue', helvetica, sans-serif">
+
+<head>
+    <meta charset="utf-8"/>
```
Comment (fcanela) [inline templates/email/components/layout-start.ejs:5]: This line gave me a little headache. The original one was `<meta charset="utf-8">`, without the trailing slash. Somehow, when getting the emails from the mocked server, the version with final slash was returned, causing some tests to break due to rendered version not matching the expected version.

File: templates/email/components/layout-start.ejs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+           <o:OfficeDocumentSettings>
+           <o:AllowPNG></o:AllowPNG>
+           <o:PixelsPerInch>96</o:PixelsPerInch>
+           </o:OfficeDocumentSettings>
+         </xml>
+      </noscript>
+<![endif]-->
+    <!--[if !mso]><!-- -->
+    <link href="https://fonts.googleapis.com/css?family=Open+Sans:400,400i,700,700i" rel="stylesheet">
+    <!--<![endif]-->
```
[the remark this answers is not in the document]
Comment (fcanela) [reply templates/email/components/layout-start.ejs:29]: I will move that feedback to our designer, but this was created with a tool specialized in creating HTML for email templates. I have to implement the design as it.

**Review by salgozino — COMMENTED** [review]
LGTM!.
After the 1inch.eth value is replaced can be merged to me.

File: templates/email/components/unsubscribe.ejs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<tr>
+    <td align="center" bgcolor="#220050" style="padding:0;Margin:0;padding-top:25px">
+        <p style="Margin:0;-webkit-text-size-adjust:none;-ms-text-size-adjust:none;mso-line-height-rule:exactly;font-family:'open sans', 'helvetica neue', helvetica, arial, sans-serif;line-height:19.2px;color:#daf0ff;font-size:16px">Best regards,<br>Kleros Zodiac Notifications Bot.</p>
+    </td>
+</tr>
+<tr>
+    <td align="center" bgcolor="#220050" style="padding:0;Margin:0;padding-top:10px;padding-bottom:25px">
+        <a href="mailto:<%- unsubscribeEmail %>?subject=Unsubscribe%20Request&body=Hello,%0D%0A%0D%0AI%20would%20like%20to%20request%20to%20unsubscribe%20from%20receiving%20further%20emails.%0D%0A%0D%0AThank%20you." target="_blank">
+            <p style="Margin:0;-webkit-text-size-adjust:none;-ms-text-size-adjust:none;mso-line-height-rule:exactly;font-family:'open sans', 'helvetica neue', helvetica, arial, sans-serif;line-height:19.2px;color:#6cc5ff;font-size:16px"><u>Unsubscribe</u></p>
```
Comment (wadader) [inline templates/email/components/unsubscribe.ejs:9]: Is this the button?

**Review by wadader — APPROVED** [review]
Looks good!

My only complaint is I was looking for an html button and couldn't find one :(

File: templates/email/components/unsubscribe.ejs (hunk 6 above)
Comment (fcanela) [reply templates/email/components/unsubscribe.ejs:9]: @wadader yes, that is the button. Honestly, the HTML is not very pleasant to read, but it is required for email client compatibility

File: templates/email/answer-issued-html.ejs (hunk 2 above)
Comment (fcanela) [reply templates/email/answer-issued-html.ejs]: Fixed in the lastest commit

**Review by jaybuidl — APPROVED** [review]

File: templates/email/components/event-links.ejs [hunk 7]
```diff
@@ -0,0 +1,57 @@
+<tr>
+    <td align="center" bgcolor="#220050" style="Margin:0;padding-bottom:5px;padding-top:25px;padding-left:25px;padding-right:25px">
+        <!--[if mso]><a href="<%- getRealityQuestionLink(notification) %>" target="_blank" hidden>
+<v:roundrect xmlns:v="urn:schemas-microsoft-com:vml" xmlns:w="urn:schemas-microsoft-com:office:word" esdevVmlButton href="<%- getRealityQuestionLink(notification) %>"
```
Comment (divyangchauhan) [inline templates/email/components/event-links.ejs:4]: can you delete commented code

**Pull request merged**